CMakeLists.txt

```diff
@@ -1762,22 +1762,20 @@ endif()
 
 if(CONFIG_BUILD_OUTPUT_EXE)
   if (NOT CONFIG_NATIVE_LIBRARY)
-    list(APPEND
-      post_build_commands
+    add_custom_target(native_executable
+      ALL
+      COMMENT "Generating ${KERNEL_EXE_NAME}"
       COMMAND
       ${CMAKE_COMMAND} -E copy ${KERNEL_ELF_NAME} ${KERNEL_EXE_NAME}
-      )
-    list(APPEND
-      post_build_byproducts
-      ${KERNEL_EXE_NAME}
-      )
-    set(BYPRODUCT_KERNEL_EXE_NAME "${PROJECT_BINARY_DIR}/${KERNEL_EXE_NAME}" CACHE FILEPATH "Kernel exe file" FORCE)
+      DEPENDS ${logical_target_for_zephyr_elf}
```

Comment:
to me the creation of `zephyr.exe` as a post build step to the zephyr executable target seems most correct, so that it is still ensured `zephyr.exe` is always created, even if a user does:
> $ ninja zephyr_final

Reply:
Thanks, changed